ribs/visualize.py

```diff
@@ -453,3 +453,145 @@ def sliding_boundaries_archive_heatmap(archive,
+    if bc_order is None:
+        cols = [key for key in archive_data.keys() if 'behavior' in key]
+        lower_bounds = archive.lower_bounds
+        upper_bounds = archive.upper_bounds
+
+    #use the requested behaviors (may be less than the original number of bcs)
+    else:
+        #check for errors in specification
+        for bc in bc_order:
+            if bc not in archive_data.keys():
+                raise ValueError(f'Behavior {bc} is not in the archive.')
+        cols = bc_order
+        #find the indices of the requested order
+        bc_indices = [int(x.split('_')[1]) for x in bc_order]
+        lower_bounds = archive.lower_bounds[bc_indices]
+        upper_bounds = archive.upper_bounds[bc_indices]
+
+    if axis_labels is not None and len(axis_labels) != len(cols):
+        raise ValueError(f'Label Mismatch: You have {len(cols)} axes'
+                         'and {len(axis_labels)} labels.')
+
+    host = plt.gca() if ax is None else ax  #which axis to plot on
+    df = archive.as_pandas(include_solutions=False)
+    vmin = np.min(df['objective']) if vmin is None else vmin
+    vmax = np.max(df['objective']) if vmax is None else vmax
+    norm = matplotlib.colors.Normalize(vmin=vmin, vmax=vmax, clip=True)
+    objectives = df['objective'].to_numpy()
+    ys = df[cols].to_numpy()
+    y_ranges = upper_bounds - lower_bounds
+
+    # transform all data to be in the first axis coordinates
+    zs = np.zeros_like(ys)
+    zs[:, 0] = ys[:, 0]
+    zs[:, 1:] = (ys[:, 1:] - lower_bounds[1:]) / \
+                y_ranges[1:] * y_ranges[0] + lower_bounds[0]
+
+    #copy the axis for the other bcs
+    axes = [host] + [host.twinx() for i in range(ys.shape[1] - 1)]
+    for i, axis in enumerate(axes):
+        axis.set_ylim(lower_bounds[i], upper_bounds[i])
+        axis.spines['top'].set_visible(False)
+        axis.spines['bottom'].set_visible(False)
+        if axis != host:
+            axis.spines['left'].set_visible(False)
+            axis.yaxis.set_ticks_position('right')
+            axis.spines["right"].set_position(("axes", i / (ys.shape[1] - 1)))
+
+    host.set_xlim(0, ys.shape[1] - 1)
+    host.set_xticks(range(ys.shape[1]))
+    labels = cols if axis_labels is None else axis_labels
+    host.set_xticklabels(labels)
+    host.tick_params(axis='x', which='major', pad=7)
+    host.spines['right'].set_visible(False)
+    host.xaxis.tick_top()
+
+    for j in range(len(archive_data)):
+        #draw straight lines between the axes in the appropriate color
+        color = matplotlib.colors.rgb2hex(cmap(norm(objectives[j])))
+        host.plot(range(ys.shape[1]),
+                  zs[j, :],
+                  c=color,
+                  alpha=alpha,
+                  linewidth=linewidth)
+
+    # Create a colorbar.
+    mappable = ScalarMappable(cmap=cmap)
+    mappable.set_clim(vmin, vmax)
+    host.figure.colorbar(mappable, pad=0.05, orientation='horizontal')
```

Comment:
Did "vertical" not work? I could see "horizontal" being a problem if we have a lot of BCs, because then the colorbar could be very long